### PR TITLE
add GetQuery, update GetFileById and GetReader

### PR DESCRIPTION
**Pull request by zubairk14**
When submitting a PR, please make sure you to:

- submit the PR to the `dev` branch, **not** the `master` branch
- in the body, reference the issue that it closes by number in the format `Closes #000`
- provide brief description of introduced enhancements

### CODE REVIEW

File: api/file.go [hunk 1]
```diff
@@ -199,7 +199,8 @@ func (file *File) UnPublish(comment string) ([]byte, error) {
 
 // GetReader gets file io.ReadCloser
 func (file *File) GetReader() (io.ReadCloser, error) {
-	endpoint := fmt.Sprintf("%s/$value", file.endpoint)
+	siteURL := file.client.AuthCnfg.GetSiteURL()
+	endpoint := fmt.Sprintf("%s/_api/web/%s/$value", siteURL, file.endpoint)
```
Comment (koltyakov) [inline api/file.go:203]: This will construct incorrect endpoint.

File: api/search.go [hunk 2]
```diff
@@ -152,6 +153,21 @@ func NewSearch(client *gosip.SPClient, endpoint string, config *RequestConfig) *
 	}
 }
 
+func (search *Search) GetQuery(query *SearchQuery) (SearchResp, error) {
```
Comment (koltyakov) [inline api/search.go:156]: Will it bling any value while providing the same exactly functionality as search.PostQuery? What are you trying to solve?

File: api/web.go [hunk 3]
```diff
@@ -352,7 +352,7 @@ func (web *Web) GetFileByPath(serverRelativeURL string) *File {
 func (web *Web) GetFileByID(uniqueID string) *File {
 	return NewFile(
 		web.client,
-		fmt.Sprintf("%s/GetFileByID('%s')", web.endpoint, uniqueID),
+		fmt.Sprintf("GetFileByID('%s')", uniqueID),
```
Comment (koltyakov) [inline api/web.go:355]: This will construct incorrect endpoint.

Comment (koltyakov) [reply api/web.go:355]: Added tests to GetFileByID/GetFolderByID methods to be sure they work. Current endpoints constructors are correct.

File: api/file.go (hunk 1 above)
Comment (koltyakov) [reply api/file.go:203]: Added test to this as well. Current implementation is correct.

**Comment by koltyakov** [comment]
@zubairk14 are you constructing endpoints from a base site?

```golang
client := &gosip.SPClient{AuthCnfg: auth} // SiteURL: https://contoso.sharepoint.com/sites/site

sp := api.NewSP(client)

data, err := sp.Web().GetFileById('{file-unique-id}').Get()
```

As chain methods take care building correct API endpoints, and the changes you're applying tell me that probably the base URL you provided is not the root of the site.

**Comment by zubairk14** [comment]
hey andrew, didn't mean to open this PR but really appreciate how quickly you responded!

i had added GetQuery() to my personal fork of your lib b/c I was running into errors with __request_metadata__ when submitting PostQuery() requests with a bearer token derived from Microsoft's Graph Oauth2 flow. 

please feel free to close/ignore this PR

**Comment by koltyakov** [comment]
@zubairk14 Do you have a sample or tech description which you're trying to solve? Just interesting, how could I help you.

**Pull request closed without merging**